molecule/core.py

```diff
@@ -198,14 +198,33 @@ def _create_templates(self):
 
         # rakefile
         kwargs = {
-            'molecule_file': self._config.molecule_file,
-            'current_platform': self._provisioner._platform,
+            'state_file': self._config.config['molecule']['state_file'],
             'serverspec_dir': self._config.config['molecule']['serverspec_dir']
         }
         utilities.write_template(
             self._config.config['molecule']['rakefile_template'],
             self._config.config['molecule']['rakefile_file'], kwargs=kwargs)
 
+    def _create_instances_state(self):
+        """
+        Creates a dict of formatted instances names and the group(s) they're
+        part of to be added to state.
+
+        :return: None
+        """
+
+        instances = {}
+        for instance in self._provisioner.instances:
+            instance_name = utilities.format_instance_name(
+                instance['name'], self._provisioner._platform,
+                self._provisioner.instances)
+            instances[instance_name] = []
+            if 'ansible_groups' in instance:
+                for group in instance['ansible_groups']:
+                    instances[instance_name].append(group)
+
+        self._state.change_state('instances', instances)
```

Comment:
I'm not entirely sold on this, but there isn't really a more elegant solution.  We could also benefit from this data living in the state file per #181.